en/autogen/README.md

```diff
@@ -0,0 +1,68 @@
+
+This approach means that we don't have to maintain a separate MAVLink implementation or API for each language.
+New features are implemented (just once) in C++, and are then automatically available in Python, Mava, etc.
+
+This topic provides an overview of the architecture and mechanisms.
+
+## Overview
+
+The common MAVLink implementation is written in C++ (it is part of MAVSDK-C++). 
+The `mavsdk_server` exposes the MAVSDK-C++ API over [gRPC](https://grpc.io/) to the language bindings.
+The API for all languages is autogenerated from a single common definition.
+
+![MAVSDK structure/architecture](../../assets/autogen/mavsdk_overview.png)
+
+The following parts are autogenerated:
+
+* MAVSDK-C++ API (i.e. header files)
+* Most of `mavsdk_server`
+* Most of the language bindings
+
+Those parts are maintained manually:
+
+* MAVSDK-C++ implementation (i.e. the MAVLink business logic)
+* New plugins need to be added to `mavsdk_server` (this may be automated in future)
+* The `System` wrapper in language bindings usually needs to be updated whenever a new plugin is added (this may be automated in future)
+
+The heart of the autogeneration system is the [MAVSDK-Proto](https://github.com/mavlink/mavsdk-proto) repository, which is described [below](#mechanisms).
+
+## Autogeneration Mechanisms {#mechanisms}
+
+The autogeneration pipeline is shown below:
+
+![Autogeneration pipeline](../../assets/autogen/autogen_overview.png)
+
+The main parts are:
+
+* [`protoc-gen-mavsdk`](https://github.com/mavlink/MAVSDK-Proto/tree/master/pb_plugins), which is a `protoc` custom plugin generating MAVSDK's code.
+* [The API definition](https://github.com/mavlink/MAVSDK-Proto/tree/master/protos), in the form of proto files.
+* Template files (per language, see e.g. the [python templates](https://github.com/mavlink/MAVSDK-Python/tree/master/other/templates/py) or the [C++ templates](https://github.com/mavlink/MAVSDK/tree/develop/templates)).
+
+
+`protoc` takes the custom plugin (`protoc-gen-mavsdk`) and the template files as inputs, and generate source code out of it.
+The way it currently works is that it generates one source file out of each `*.proto` file. For instance, `action.proto` becomes `Action.java` in MAVSDK-Java.
+
+In some languages (typically C++), we need to generate multiple source files out of one proto definition file, and this is the reason why the C++ generation script runs protoc [in](https://github.com/mavlink/MAVSDK/blob/develop/tools/generate_from_protos.sh#L72-L74) [multiple](https://github.com/mavlink/MAVSDK/blob/develop/tools/generate_from_protos.sh#L76-L77) [passes](https://github.com/mavlink/MAVSDK/blob/develop/tools/generate_from_protos.sh#L79-L81), once [for each template](https://github.com/mavlink/MAVSDK/blob/develop/tools/generate_from_protos.sh#L79-L81) (e.g. "templates/plugin_h" defines the templates to generate `action.h`, `telemetry.h`, ..., and "templates/plugin_cpp" is responsible for `action.cpp`, `telemetry.cpp`, ...).
```

Comment:
I would use permalinks here, otherwise the line numbers will be wrong in the future.